lib/src/input_validation/validate_function.dart

```diff
@@ -40,23 +39,151 @@ class ValidateFunction {
+        valid = true;
+    }
+
+    return valid;
+  }
+
+  bool checkValues(List<String> matrix1Values, List<String> matrix2Values){
+    for(int i = 0; i < matrix1Values.length; i++){
+      String token = _sanitizeToken(matrix1Values[i]);
+      if(!Pattern.validOperand.hasMatch(token))
+        return false;
+    }
+
+    for(int i = 0; i < matrix2Values.length; i++) {
+      String token = _sanitizeToken(matrix2Values[i]);
+      if (!Pattern.validOperand.hasMatch(token))
+        return false;
+    }
+
+    return true;
+  }
+
+  bool validateOperator(String value){
+    if(Pattern.validOperator.hasMatch(value))
+      return true;
+    else
+      return false;
+  }
+
+  bool validateSize(List<String> input){
+    String matrix1 = input[0].replaceAll("&", "");
+    String operator = input[1];
+    String matrix2 = input[2].replaceAll("&", "");
+
+    if(Pattern.addSubtractOperator.hasMatch(operator)){
+      if(isRowSameSize(matrix1, matrix2) && isColumnSameSize(matrix1, matrix2))
+        return true;
+      else
+        return false;
+    }
+    else if(Pattern.multiplyDivideOperator.hasMatch(operator)){
+      if(isRowColSameSize(matrix1, matrix2))
+        return true;
+      else
+        return false;
+    }
+    else
+      return false;
+  }
+
+  bool isRowSameSize(String matrix1, String matrix2){
+    List<String> rowsMatrix1 = matrix1.split("!");
+    List<String> rowsMatrix2 = matrix2.split("!");
+
+    if(rowsMatrix1.length == rowsMatrix2.length)
+      return true;
+    else
+      return false;
+  }
+
+  bool isColumnSameSize(String matrix1, String matrix2){
+    List<String> rowsMatrix1 = matrix1.split("!");
+    List<String> rowsMatrix2 = matrix2.split("!");
+
+    //need to check if correct number of columns exist in a row for matrix
+    if(isValidColumns(rowsMatrix1) && isValidColumns(rowsMatrix2)) {
+      List<String> colMatrix1 = rowsMatrix1[0].split(",");
+      List<String> colMatrix2 = rowsMatrix2[0].split(",");
+
+      if (colMatrix1.length == colMatrix2.length)
+        return true;
+      else
+        return false;
+    }
+    else
+      return false;
+  }
+
+  bool isRowColSameSize(String matrix1, String matrix2){
+    List<String> rowsMatrix2 = matrix2.split("!");
+    List<String> rowsMatrix1 = matrix1.split("!");
+
+    //need to check if correct number of columns exist in a row for matrix
+    if(isValidColumns(rowsMatrix1) && isValidColumns(rowsMatrix2)) {
+      List<String> colMatrix1 = rowsMatrix1[0].split(",");
+
+      if (rowsMatrix2.length == colMatrix1.length)
+        return true;
+      else
+        return false;
+    }
+    else
+      return false;
+  }
+
+  bool isValidColumns(List<String> rowsMatrix){
+    int colSize = rowsMatrix[0].split(",").length;
+
+    for (int i=0; i < rowsMatrix.length; i++) {
+      List<String> colMatrix = rowsMatrix[i].split(",");
+      if (colSize != colMatrix.length)
+        return false;
```

Comment:
try to allow loops to run to completion, it makes code more readable

Reply:
I don't think it makes sense to go through a complete for loop if one item fails so leaving is as is.